### PR TITLE
Last errors

### PR DESCRIPTION
**Pull request by christianblais**
This allows the recording of (mostly) user/configuration errors happening while executing user defined handlers.

Screen Shot 2022-02-10 at 12 23 27 PM

Screen Shot 2022-02-10 at 12 32 21 PM

### CODE REVIEW

File: test/fixtures/handlers.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ email_deliver:
...
   service_name: 'Handlers::Emailer'
   settings:
-    recipients: "test+to@example.com"
-    from: "test+from@example.com"
-    subject: "New customer registered"
-    body: "New customer registered from Canada with id {{ id }} and email {{ email }}."
+    :recipients: "test+to@example.com"
+    :from: "test+from@example.com"
+    :subject: "New customer registered"
+    :body: "New customer registered from Canada with id {{ id }} and email {{ email }}."
```
Comment (christianblais) [inline test/fixtures/handlers.yml:8]: We persist them as symbols, so this is just for the fixture to mimic what's happening in real life. I think?

**Review by tjoyal — COMMENTED** [review]
Discussion to be had on the product side:

One of the problems I see with only tracking errors is that they do not communicate to the end user "it is working" which I think has as much value.

I see two paths forward:

Either we shouldn't be tracking "last errors" only but rather "last events" which could be either a success or an error.

Alternatively, we could track “last errors” + “last success” in a separate column. That way we could list the last X errors & communicate if the handler is working or not.

Similar product outcome, distinct implementations.

I am particularly interested in “hiding away” at some point or communicating that an error is no longer a problem. We can in both cases. Either “last events” will push the errors out, or we can have the “last success” flag the “last errors” as not the latest.

I think I prefer “last events” as it presents a more simple mental model for the end user and the UI vs the 2 columns who end up with more granular data but could confuse the user. 

I am worried of an intermittent error or a past configuration mistake to be permanently visible, so we might also need to look at adding a “clearing” endpoint for “last errors”?

**Comment by christianblais** [comment]
As per our discussion, closing in favor of last interactions.

**Comment by tjoyal** [comment]
Solved via https://github.com/christianblais/triggerify/pull/116

**Pull request closed without merging**